backend/src/main/java/com/praksa/auction/config/security/services/UserDetailsServiceImpl.java

```diff
@@ -0,0 +1,20 @@
+package com.praksa.auction.config.security.services;
+
+import com.praksa.auction.model.Person;
+import com.praksa.auction.repository.PersonRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+@Service
+public class UserDetailsServiceImpl implements UserDetailsService{
+    @Autowired
+    PersonRepository personRepository;
+    @Override
+    public UserDetails loadUserByUsername(String email) throws UsernameNotFoundException {
+        Person person = personRepository.findByEmail(email).orElseThrow(() -> new UsernameNotFoundException("User with email " + email +" was not found!"));
```

Comment:
We said we do not get this error message. Can we throw just simple exepction here then, instead of  UsernameNotFoundException?

Reply:
Since this loadUserByUsername method is overridden from UserDetailsServiceImpl, it has to throw UsernameNotFoundException if it is throwing exceptions. The only alternative that I could come up with in order to get this message was to check if email exists before the authentication begins in contorller and in that case, I would delete this orElseThrow part of method. Would that be good solution?